clients/src/main/java/org/apache/kafka/clients/ManualMetadataUpdater.java

```diff
@@ -69,6 +70,11 @@ public void handleDisconnection(String destination) {
     }
 
     @Override
+    public void handleAuthenticationFailure(AuthenticationException exception) {
+        // Do nothing
```

Comment:
Not sure if it is in scope, but are we not handling authentication failures between brokers? Maybe we should at least log a warning or something?

Reply:
We don't fail the broker on authentication failures, but there is sufficient information in the broker logs to identify the failure. I think this log entry should be a `debug` rather than a `warn` since we already have an error log entry, so the warning doesn't add much value. With this change the log looks like:
```
ERROR [Controller id=0, targetBrokerId=0] Connection to node 0 failed authentication due to: Authentication failed due to invalid credentials with SASL mechanism PLAIN (org.apache.kafka.clients.NetworkClient)
WARN An authentication error occurred in broker-to-broker communication! (org.apache.kafka.clients.ManualMetadataUpdater)
```

Reply:
Agreed. And we should add a comment explaining that.